prow/gerrit/adapter/adapter.go

```diff
@@ -168,6 +169,43 @@ func createRefs(reviewHost string, change client.ChangeInfo, cloneURI *url.URL,
 	return refs, nil
 }
 
+// failedJobs find jobs currently reported as failing (used for retesting).
+//
+// Failing means the job is complete and not passing.
+// Scans messages for prow reports, which lists jobs and whether they passed.
+// Job is included in the set if the latest report has it failing.
+func failedJobs(account int, revision int, messages ...gerrit.ChangeMessageInfo) sets.String {
+	failures := sets.String{}
+	times := map[string]time.Time{}
+	for _, message := range messages {
+		if message.Author.AccountID != account { // Ignore reports from other accounts
+			continue
+		}
+		if message.RevisionNumber != revision { // Ignore reports for old commits
```

Comment:
I'm not quite sure of the UX for this case, technically it's possible to have a use case where you run an optional job `/test foo` - > (it fails)  -> you send a couple of new patchsets/revisions and then want to run it again. 
Do we care if `/retest` should honor it or should we expect another `/test foo` on the newer revision?
How do we handle this in GitHub?